dev/create-release/release-build.sh

```diff
@@ -95,6 +95,28 @@ if [ -z "$SPARK_VERSION" ]; then
     | grep -v INFO | grep -v WARNING | grep -v Download)
 fi
 
+# Verify we have the right java version set
+java_version=$("${JAVA_HOME}"/bin/javac -version 2>&1 | cut -d " " -f 2)
```

Comment:
@holdenk, should we maybe catch the case when `JAVA_HOME` is missing too?

If so, I think we could do something like ...

```bash
if [ -z "$JAVA_HOME" ]; then
  echo "Please set JAVA_HOME."
  exit 1
fi
...
```

Or maybe...

```bash
if [[ -x "$JAVA_HOME/bin/javac" ]]; then
  javac_cmd="$JAVA_HOME/bin/javac"
else
  javac_cmd=javac
fi

java_version=$("$javac_cmd" -version 2>&1 | cut -d " " -f 2)
...
```

I tested both in my local.

Reply:
Thanks. The logic we use in make-distribution to resolve JAVA_HOME when not set is a bit more complicated, but I think for the release script its reasonable (and probably a good practice) to explicitly require a JAVA_HOME to be set so I'll go with the first of the 2.